tuning/config/acceleration_configs/attention_and_distributed_packing.py

```diff
@@ -0,0 +1,35 @@
+# Standard
+from dataclasses import dataclass
+
+# Local
+from .utils import ensure_nested_dataclasses_initialized, parsable_dataclass
+
+
+@parsable_dataclass
+@dataclass
+class PaddingFree:
+
+    method: str = "huggingface"
+
+    def __post_init__(self):
+        if self.method != "huggingface":
+            raise ValueError("only 'huggingface' method currently supported.")
```

Comment:
Can you describe what "huggingface" method padding free means compared to other methods?

Reply:
"huggingface" means the recent PR made by the team https://github.com/huggingface/transformers/pull/31629 that uses FA2. I left it extensible in the potential event that we want to support other variants of padding free attention (e.g., native torch SDPA with nested tensors). No concrete plans now, but future proofing.